src/mono/wasm/build/WasmApp.Common.targets

```diff
@@ -334,6 +334,108 @@
+      <_WasmPropRequiresFullNativeRebuild Include="WasmAllowUndefinedSymbols=$(WasmAllowUndefinedSymbols)" />
+      <_WasmPropRequiresFullNativeRebuild Include="EnableAggressiveTrimming=$(EnableAggressiveTrimming)" />
+      <_WasmPropRequiresFullNativeRebuild Include="PublishTrimmed=$(PublishTrimmed)" />
+      <_WasmPropRequiresFullNativeRebuild Include="_ExtraTrimmerArgs=$(_ExtraTrimmerArgs)" />
+      <_WasmPropRequiresFullNativeRebuild Include="UsingMicrosoftNETSdkWebAssembly=$(UsingMicrosoftNETSdkWebAssembly)" />
+
+      <!-- items+values -->
+      <_WasmItemRequiresFullNativeRebuild Include="@(RuntimeHostConfigurationOption->'%(Identity)=%(Value)')" />
+    </ItemGroup>
+
+    <!-- FIXME: ideally we would also compute hash of the TrimmerRootDescriptor files -->
+    <Hash ItemsToHash="@(_WasmPropRequiresFullNativeRebuild);
+                       @(_WasmItemRequiresFullNativeRebuild);
+                       @(WasmAssembliesToBundle);
+                       @(GenerateRuntimeConfigurationFilesInputs);
+                       @(WasmNativeAsset);
+                       @(NativeFileReference);
+                       @(LinkerArg);
+                       @(TrimmerRootDescriptor);
+                       @(EmccExportedRuntimeMethod);
+                       @(EmccExportedLibraryFunction);
+                       @(EmccExportedFunction);
+                       @(MonoAotCrossCompiler);
+                       @(_MonoComponent);
+                       ">
+      <Output TaskParameter="HashResult" PropertyName="_WasmNativePropertiesCurrentHash" />
+    </Hash>
+
+    <ItemGroup>
+      <!-- cleanup -->
+      <_WasmPropRequiresFullNativeRebuild Remove="@(_WasmPropRequiresFullNativeRebuild)" />
+      <_WasmItemRequiresFullNativeRebuild Remove="@(_WasmItemRequiresFullNativeRebuild)" />
+    </ItemGroup>
+
+    <ReadLinesFromFile File="$(IntermediateOutputPath)WasmNativePropertiesHash.stamp" Condition="Exists('$(IntermediateOutputPath)WasmNativePropertiesHash.stamp')">
+      <Output TaskParameter="Lines" PropertyName="_WasmNativePropertiesPreviousHash" />
+    </ReadLinesFromFile>
+
+    <PropertyGroup>
+      <_WasmNativePropertiesDiffer Condition="'$(_WasmNativePropertiesPreviousHash)' != '$(_WasmNativePropertiesCurrentHash)'">true</_WasmNativePropertiesDiffer>
+    </PropertyGroup>
+
+    <RemoveDir Directories="$(_WasmIntermediateOutputPath)" 
+      Condition="Exists('$(_WasmIntermediateOutputPath)') and ('$(_WasmNativePropertiesDiffer)' == 'true')"
+    />
+    <MakeDir Directories="$(_WasmIntermediateOutputPath)" 
+      Condition="!Exists('$(_WasmIntermediateOutputPath)')" />
+
+    <WriteLinesToFile Lines="$(_WasmNativePropertiesCurrentHash)"
+      Condition="'$(_WasmNativePropertiesDiffer)' == 'true'"
+      File="$(IntermediateOutputPath)WasmNativePropertiesHash.stamp" 
+      Overwrite="true" 
+      WriteOnlyWhenDifferent="true" />
```

Comment:
this one seems redundant since we already check _WasmNativePropertiesDiffer==true